<details><summary>Failed to submit graph for 71c08dfcdf09e6f44348d25005c0e8bbf6c1f244
</summary>Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmp3mqwd714/botocore-feedstock.yaml --custom-metadata-file /tmp/tmp3mqwd714/custom_metadata.yaml --release-name botocore-feedstock-67-71c08dfcdf09e6f44348d25005c0e8bbf6c1f244 botocore-feedstock/recipe', with error: 'INFO:root:Found feedstock botocore-feedstock at /tmp/tmp3mqwd714/botocore-feedstock/recipe
INFO:root:Build/Test graph from botocore-feedstock
Created by sbx_user1051 on 2025-12-03 14:19:21
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmp3mqwd714/botocore-feedstock.yaml --custom-metadata-file /tmp/tmp3mqwd714/custom_metadata.yaml --release-name botocore-feedstock-67-71c08dfcdf09e6f44348d25005c0e8bbf6c1f244 botocore-feedstock/recipe

INFO:root:Found abs.yaml in /tmp/tmp3mqwd714/botocore-feedstock
INFO:pbpcli.utils.common:Setting env var: ANACONDA_ROCKET_ENABLE_PY314 = True.
INFO:root:Creating graph from feedstock at: botocore-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.09 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 3 unchanging vars:(396), changing vars (3): ['blas_impl', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
WARNING: Number of parsed outputs does not match detected raw metadata blocks. Identified output block may be wrong! If you are using Jinja conditionals to include or exclude outputs, consider using `skip: true  # [condition]` instead.
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
INFO:pbpcli.utils.common:Unsetting env var: ANACONDA_ROCKET_ENABLE_PY314
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 2097, in _get_contents
    rendered = template.render(environment=env)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/jinja2/environment.py", line 1295, in render
    self.environment.handle_exception()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/jinja2/environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/tmp/tmp3mqwd714/botocore-feedstock/recipe/meta.yaml", line 97, in top-level template code
    - pip check
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/jinja2/async_utils.py", line 48, in wrapper
    return normal_func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/jinja2/filters.py", line 612, in sync_do_join
    return str(d).join(map(str, value))
                       ~~~^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'ignore_tests_list' is undefined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 413, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 624, in from_feedstock
    build_configs, test_configs = expand_build_matrix(feedstock.recipe_dir_path, src_channels)
                                  ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 79, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 544, in variants_for_feedstock_and_platform
    metadata_tuples = api.render(
        parsed.recipe,
    ...<2 lines>...
        variants=None,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/api.py", line 68, in render
    return render_metadata_tuples(
        metadata_tuples,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 1097, in render_metadata_tuples
    for od, om in meta.get_output_metadata_set(
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        permit_undefined_jinja=not finalize,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bypass_env_check=bypass_env_check,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ):
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 2781, in get_output_metadata_set
    conda_packages = finalize_outputs_pass(
        ref_metadata,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1072, in finalize_outputs_pass
    parent_metadata.parse_until_resolved()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1434, in parse_until_resolved
    self.parse_again(
    ~~~~~~~~~~~~~~~~^
        permit_undefined_jinja=False,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        allow_no_other_outputs=allow_no_other_outputs,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bypass_env_check=bypass_env_check,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1339, in parse_again
    self._get_contents(
    ~~~~~~~~~~~~~~~~~~^
        permit_undefined_jinja,
        ^^^^^^^^^^^^^^^^^^^^^^^
        allow_no_other_outputs=allow_no_other_outputs,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bypass_env_check=bypass_env_check,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ),
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 2107, in _get_contents
    raise CondaBuildUserError(
        f"Failed to render jinja template in {self.meta_path}:\n{str(ex)}"
    )
conda_build.exceptions.CondaBuildUserError: Failed to render jinja template in /tmp/tmp3mqwd714/botocore-feedstock/recipe/meta.yaml:
'ignore_tests_list' is undefined
'
aws lambda request id: 4bd9ce81-0cd3-5b56-bde0-af2b45f6fc1d
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%604bd9ce81-0cd3-5b56-bde0-af2b45f6fc1d%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -38,8 +38,8 @@ requirements:
 {% set ignore_tests = " --ignore=tests/integration/" %}
 
 # tests/functional/leak/test_resource_leaks.py:39: AssertionError
-{% set ignore_tests_list = "" %}
-{% set ignore_tests_list = " test_create_single_client_memory_constant" %}  # [py==314 and osx]
+{% set ignore_tests_list = [] %}
+{% set ignore_tests_list = ignore_tests_list + ["test_create_single_client_memory_constant"] %}  # [py==314 and osx]
 
 test:
   source_files:
@@ -98,7 +98,7 @@ test:
     - pip check
     # Disabled win tests. It takes too much time—around 5 hours.
     # https://github.com/AnacondaRecipes/botocore-feedstock/pull/60#discussion_r1933937673
-    - pytest -ra -v tests {{ ignore_tests }} -k "not slow and not ({{ ignore_tests_list | join(' or ') }})" # [not win]
+    - pytest -ra -v tests {{ ignore_tests }} -k "not slow{% if ignore_tests_list %} and not ({{ ignore_tests_list | join(' or ') }}){% endif %}"  # [not win]
   requires:
     - pip
     - pytest >=8.1.1

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -20,7 +20,14 @@ requirements:
     - python
     - setuptools >=71.1.0
     - wheel >=0.43.0
-    - packaging
+    # PBP/Taskcluster-orcestration issue
+    # this is workaround pinning packaging 24.1
+    # Executing command: conda create -n testenv --dry-run --json --quiet 
+    # --strict-channel-priority --override-channels -c file:///home/task_176477320693380/temp_channel/ 
+    # -c pbp_src_multichannel botocore python=3.12
+    # Error: Error: Package botocore not found in the fetched packages.
+    - packaging 24.1  # [py==312 and linux]
+    - packaging       # [not (py==312 and linux)]
   run:
     - python
     - jmespath >=0.7.1,<2.0.0

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -38,7 +38,8 @@ requirements:
 {% set ignore_tests = " --ignore=tests/integration/" %}
 
 # tests/functional/leak/test_resource_leaks.py:39: AssertionError
-{% set ignore_tests_list = "test_create_single_client_memory_constant" %} # [py==314 and osx]
+{% set ignore_tests_list = "" %}
+{% set ignore_tests_list = " test_create_single_client_memory_constant" %}  # [py==314 and osx]
 
 test:
   source_files:

